pkg/epinio/detail/applications.vue

```diff
@@ -111,23 +113,32 @@ export default Vue.extend<Data, any, any, any>({
 
       this.gitSource = GitUtils[this.gitType].normalize.repo(res);
 
-      const commit = this.value.appSourceInfo?.details.filter((ele: { label: string; }) => ele.label === 'Revision')[0]?.value;
-
-      if (commit) {
-        this.gitDeployment.deployedCommit = {
-          short: commit?.slice(0, 7),
-          long:  commit
-        };
-      }
-
       await this.fetchCommits();
     },
     async fetchCommits() {
-      const { usernameOrOrg, repo, branch } = this.value.appSource.git;
+      const {
+        usernameOrOrg, repo, branch, commit
+      } = this.value.appSource.git;
+
+      if (branch?.name) {
+        this.gitDeployment.commits = await this.$store.dispatch(`${ this.gitType }/fetchCommits`, {
+          username: usernameOrOrg, repo, branch
+        });
+      } else {
+        // branch name is missing, application is create by revision -> shows the revision commit
+        this.gitDeployment.commits = await this.$store.dispatch(`${ this.gitType }/fetchRevisionCommit`, {
+          username: usernameOrOrg, repo, revision: commit
```

Comment:
This unfortunately won't work. The revision when deploying just via `git` is a core revision. At the moment it can be either a commit or branch. I think Enrico was also saying it can now be a tag as well.

If it's a branch we won't know what commit it was deployed from (depends on `head` of that branch when the push command was run).

Overall this solution is nice, but it goes further than what i thought we'd agreed. Basically we go through the config options provided (provider --> user/org --> branch --> commit)... if there's no branch we stop at that point and don't progress to commit. As the basic `git` application, even with the `provider` populated, doesn't contain a `branch` we stop there and don't need to work out what revision is.

Reply:
I didn't consider that revision could depend also from `head` and I assumed the last commit would be the valid one , in case of missing branch.
I will left code improvements and simply leave blank when branch is missing.